src/rsz/src/Rebuffer.cc

```diff
@@ -136,6 +131,77 @@ void pruneCapVsAreaOptions(StaState* sta, BufferedNetSeq& options)
   options.resize(si);
 }
 
+static const RiseFallBoth* transitionGcd(const RiseFallBoth* a,
+                                         const RiseFallBoth* b)
+{
+  if (a == b) {
+    return a;
+  }
+  if (a == nullptr) {
+    return b;
+  }
+  if (b == nullptr) {
+    return a;
+  }
+  return RiseFallBoth::riseFall();
+}
+
+void RepairSetup::annotateLoadSlacks(BufferedNetPtr& bnet, Vertex* root_vertex)
+{
+  using BnetType = BufferedNetType;
+  using BnetPtr = BufferedNetPtr;
+
+  for (auto rf_index : RiseFall::rangeIndex()) {
+    arrival_paths_[rf_index].init();
+  }
+
+  visitTree(
+      [&](auto& recurse, int level, const BnetPtr& bnet) -> int {
+        switch (bnet->type()) {
+          case BnetType::wire:
+          case BnetType::buffer:
+            return recurse(bnet->ref());
+          case BnetType::junction:
+            return recurse(bnet->ref()) + recurse(bnet->ref2());
+          case BnetType::load: {
+            const Pin* load_pin = bnet->loadPin();
+            Vertex* vertex = graph_->pinLoadVertex(load_pin);
+            PathRef req_path
+                = sta_->vertexWorstSlackPath(vertex, sta::MinMax::max());
+            PathRef arrival_path = req_path;
+
+            while (!req_path.isNull()
+                   && arrival_path.vertex(sta_) != root_vertex) {
+              TimingArc* arc;
+              PathRef next_path;
+              arrival_path.prevPath(sta_, next_path, arc);
+              arrival_path = next_path;
+              assert(!arrival_path.isNull());
+            }
+
+            if (arrival_path.isNull()) {
+              bnet->setSlackTransition(nullptr);
+              bnet->setSlack(INF);
+            } else {
+              const RiseFall* rf = req_path.transition(sta_);
+              bnet->setSlackTransition(rf->asRiseFallBoth());
+              bnet->setSlack(req_path.required(sta_)
+                             - arrival_path.arrival(sta_));
+
+              if (arrival_paths_[rf->index()].isNull()) {
+                arrival_paths_[rf->index()] = arrival_path;
+              }
+            }
+
+            return 1;
+          }
+          default:
+            abort();
```

Comment:
Do not use abort - an error suffices.

Reply:
Error as in `logger_->error()`? That's more effort with writing the message and picking the number but I'll try to do that if that's expected

Reply:
This leaves the user with a sudden exit and no explanation at all.  I think the effort is small to put in an appropriate logger_->error()

Reply:
Got it

Reply:
(To be clear this branch should only be taken if an internal invariant is violated, otherwise I’d never use abort.)

Reply:
Better is to have exhaustive case statements and no default at all.  The compiler will error if you miss a case.